service-netty/src/main/java/info/bitrich/xchangestream/service/netty/NettyStreamingService.java

```diff
@@ -67,6 +68,7 @@ public Subscription(ObservableEmitter<T> emitter, String channelName, Object[] a
     private final NioEventLoopGroup eventLoopGroup;
     protected Map<String, Subscription> channels = new ConcurrentHashMap<>();
     private boolean compressedMessages = false;
+    private List<ObservableEmitter<Throwable>> reconnFailEmitters = new LinkedList<>();
```

Comment:
My comment below makes this moot, but you should use a concurrency-safe list implementation like `LinkedBlockingQueue`, otherwise you're likely to get `ConcurrentModificationException` if the list is modified while it is being iterated.